src/Types/ReceiveName.cs

```diff
@@ -2,5 +2,47 @@ namespace Concordium.Sdk.Types;
 
 /// <summary>
 /// A receive name of the contract function called. Expected format: "&lt;contract_name&gt;.&lt;func_name&gt;".
+/// The name must not exceed 100 bytes and all characters are ascii alphanumeric or punctuation.
 /// </summary>
-public readonly record struct ReceiveName(string Receive);
+public sealed record ReceiveName
+{
+    private const uint MaxByteLength = 100;
+
+    /// <summary>
+    /// Name with format "&lt;contract_name&gt;.&lt;func_name&gt;".
+    /// </summary>
+    public string Receive { get; init; }
+
+    internal ReceiveName(string receive) => this.Receive = receive;
+
+    /// <summary>
+    /// Try parse input name against expected format.
+    /// </summary>
+    /// <param name="name">Input receive name.</param>
+    /// <param name="receiveName">Parsed receive name when succeeded.</param>
+    /// <returns>True if name satisfied expected format.</returns>
+    public static bool TryParse(string name, out ReceiveName? receiveName)
+    {
+        var validate = IsValid(name);
+        receiveName = validate ? new ReceiveName(name) : null;
+        return validate;
+    }
+
+    private static bool IsValid(string name)
+    {
+        if (!name.Contains('.'))
+        {
+            return false;
```

Comment:
Ideally, we would provide the user with the reason why it was invalid.
Not sure what the C# way of doing this is

Reply:
Generally you would throw exceptions, but this is very exceptions for the GC which is why I’m in favor of the Try pattern. The pros with the try pattern is you avoid the expense exception allocated, but on the other hand you have null references where the user needs to check which one is not null.

I have added error types to the output.